### ISSUE TITLE
Search functionality on the frontend

### ISSUE THREAD

**Issue by Shadyzpop** [SocialXNetwork/socialx_react_native#146]
when searching for someone we fire `SEARCH_FOR_PROFILES` which removes the current queue on the store then `SYNC_SEARCH_FOR_PROFILES` comes back afterwards to either populate the search store or it will return nothing which is also fine

the problems:

# first problem
on multiple input changes we debounce the search which sometimes clusters the function calls if typed fast enough, it will call `SEARCH_FOR_PROFILES` twice then the first `SYNC_SEARCH_FOR_PROFILES` comes back with data but when the second one comes it will trigger the `setGlobal` to remove the prev aliases, but  if the first and second sync have the same user, it will basically remove that user from the store and wont display it on the search results

# second problem
if we search for someone and return their data then continuously remove the search input it will cause a problem while firing `setGlobal({alisesToRemove})` on action
```
{
  type: 'data/profiles/SEARCH_FOR_PROFILES',
  payload: [ <== populated
    ...aliases
  ]
}
```

which will turn into a `cannot read property 'fullName' of undefined` because it tries to reference a profile from the store but its already removed?

the `SYNC_SEARCH_FOR_PROFILES` is not returning data afterwards we call the above action

**Comment by alexcroitoru95** [comment]
Yup, I can agree.

### Steps to reproduce for first bug

1. Write the first 3 letters of a fullname from any user
2. Wait for the response to appear
3. Add '@' at the end of the search input
4. Voila ('**fullName of undefined**')

### Steps to reproduce for second bu
1. Write the first 3 letters of a fullname from any user 
2. Wait for the response to appear
3. Delete slowly the letters one by one
4. Voila ('**fullName of undefined**')

**Comment by alexsirbu04** [comment]
Fixed in a80f496e1e09994f4da7f964e4a63481deb74ad5.
